### PR TITLE
[frontend/backend] Display related Malware-analyses in Obs overview (#3478)

### LINKED ISSUE #3478: Modifying observables overview tab to display Malware Analyses
## Description

Following the completion of https://github.com/OpenCTI-Platform/opencti/issues/1565, the next step is:
- [x] Modify File/URL/NetworkTraffic/Artifact/Domain/Hostname Overview Tab to display the list of related Malware-analysis
- [x] Modify [[Hybrid Analysis enrichement connector](https://github.com/OpenCTI-Platform/connectors/tree/master/internal-enrichment/hybrid-analysis-sandbox)](https://github.com/OpenCTI-Platform/connectors/tree/master/internal-enrichment/hybrid-analysis-sandbox) to have a means to feed the platform with the newly Malware-Analysis SDO. This connector will now (in addition of what it is currently doing):
  - create a Malware-Analysis SDO with properties filled in
  - add the enriched SCOs (File, URL or Network Traffic) as a Nested Object of the Malware-Analysis instance
add every SCOs created and related through the enrichment process as Nested Objects of the Malware-Analysis instance.
- [ ] Write a blog article

### PR DESCRIPTION
**Pull request by yassine-ouaamou**
### Proposed changes

* Added a Malware Analyses list in the overview of some observables (File, Url, Hostname and Domain name)

image

### Related issues

* https://github.com/OpenCTI-Platform/opencti/issues/3478
*
### How to get the list?
1. Create an observable (File, Url, Hostname and Domain name)
2. Create malware analyses with a nested object "sample" pointing the observable

### Checklist

- [x] I consider the submitted work as finished
- [x] I tested the code for its functionality
- [ ] I wrote test cases for the relevant uses case
- [ ] I added/update the relevant documentation (either on github or on notion)
- [x] Where necessary I refactored code to improve the overall quality

### Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: opencti-platform/opencti-front/src/private/components/observations/stix_cyber_observables/StixCyberObservableMalwareAnalyses.tsx [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+    ],
+  };
+
+  const queryRef = useQueryLoading<StixCyberObservableMalwareAnalysesQuery>(query, options);
+  const { t } = useFormatter();
+
+  return (
+    <div style={{ marginTop: 20 }}>
+      <Typography variant="h3" gutterBottom={true} style={{ float: 'left' }}>
+        {t('Malware Analyses run on this observable')}
```
Comment (Kedae) [inline opencti-platform/opencti-front/src/private/components/observations/stix_cyber_observables/StixCyberObservableMalwareAnalyses.tsx]: Missing localization

File: opencti-platform/opencti-front/src/private/components/observations/stix_cyber_observables/StixCyberObservableMalwareAnalyses.tsx [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+            >
+              {t('No malware analyses about this entity.')}
+            </span>
+      </div>
+    );
+  }
+
+  return (
+    <>
+      {malwareAnalyses?.edges?.map((malwareAnalysisEdge, index) => <StixCyberObservableMalwareAnalysesListLine key={index}
```
Comment (Kedae) [inline opencti-platform/opencti-front/src/private/components/observations/stix_cyber_observables/StixCyberObservableMalwareAnalyses.tsx]: Can you indent properly ?

File: opencti-platform/opencti-front/src/private/components/observations/stix_cyber_observables/StixCyberObservableMalwareAnalyses.tsx [hunk 3]
```diff
@@ -0,0 +1,287 @@
...
+        }}
+      >
+            <span
+              style={{
+                display: 'table-cell',
+                verticalAlign: 'middle',
+                textAlign: 'center',
+              }}
+            >
+              {t('No malware analyses about this entity.')}
```
Comment (Kedae) [inline opencti-platform/opencti-front/src/private/components/observations/stix_cyber_observables/StixCyberObservableMalwareAnalyses.tsx]: Missing Localization

File: opencti-platform/opencti-front/src/private/components/observations/stix_cyber_observables/StixCyberObservableMalwareAnalyses.tsx [hunk 4]
```diff
@@ -0,0 +1,287 @@
...
+};
+
+const StixCyberObservableMalwareAnalysesList: FunctionComponent<StixCyberObservableMalwareAnalysesListProps> = ({ queryRef }) => {
+  const { malwareAnalyses } = usePreloadedQuery<StixCyberObservableMalwareAnalysesQuery>(query, queryRef);
+  const { t } = useFormatter();
+  if ((malwareAnalyses?.edges ?? []).length === 0) {
+    return (
+      <div
+        style={{
+          display: 'table',
```
Comment (Kedae) [inline opencti-platform/opencti-front/src/private/components/observations/stix_cyber_observables/StixCyberObservableMalwareAnalyses.tsx]: Can you add class ?

File: opencti-platform/opencti-front/src/private/components/observations/stix_cyber_observables/StixCyberObservableMalwareAnalyses.tsx [hunk 5]
```diff
@@ -0,0 +1,287 @@
...
+          height: '100%',
+          width: '100%',
+          paddingTop: 15,
+          paddingBottom: 15,
+        }}
+      >
+            <span
+              style={{
+                display: 'table-cell',
```
Comment (Kedae) [inline opencti-platform/opencti-front/src/private/components/observations/stix_cyber_observables/StixCyberObservableMalwareAnalyses.tsx]: Same

File: opencti-platform/opencti-front/src/private/components/observations/stix_cyber_observables/StixCyberObservableDetails.jsx [hunk 6]
```diff
@@ -146,16 +148,20 @@ const StixCyberObservableDetailsComponent = ({ stixCyberObservable }) => {
...
               </Grid>
             );
           })}
         </Grid>
-        <Divider />
+        <Divider/>
         <StixCyberObservableIndicators
           stixCyberObservable={stixCyberObservable}
         />
+        {isAnalysableObservable && <StixCyberObservableMalwareAnalyses
```
Comment (Kedae) [inline opencti-platform/opencti-front/src/private/components/observations/stix_cyber_observables/StixCyberObservableDetails.jsx]: Can you indent properly ?

File: opencti-platform/opencti-front/src/private/components/analysis/malware_analyses/MalwareAnalysisDetails.tsx [hunk 7]
```diff
@@ -92,15 +92,15 @@ const MalwareAnalysisDetails: FunctionComponent<MalwareAnalysisDetailsProps> = (
...
+                            {t('Report name')}
                         </Typography>
                             <ExpandableMarkdown
                                 source={data.result_name}
                                 limit={300}
                             />
                         <Typography variant="h3" gutterBottom={true} style={{ marginTop: 20 }}
                         >
-                            {t('Malware result')}
+                            {t('Maliciousness')}
```
Comment (Kedae) [inline opencti-platform/opencti-front/src/private/components/analysis/malware_analyses/MalwareAnalysisDetails.tsx]: Missing localization

File: opencti-platform/opencti-front/src/private/components/analysis/malware_analyses/MalwareAnalysisDetails.tsx [hunk 8]
```diff
@@ -92,15 +92,15 @@ const MalwareAnalysisDetails: FunctionComponent<MalwareAnalysisDetailsProps> = (
                                 limit={300}
                             />
                         <Typography variant="h3" gutterBottom={true} style={{ marginTop: 20 }}>
-                            {t('Result name')}
+                            {t('Report name')}
```
Comment (Kedae) [inline opencti-platform/opencti-front/src/private/components/analysis/malware_analyses/MalwareAnalysisDetails.tsx]: Why report name when the data is result_name ?

Comment (yassine-ouaamou) [reply opencti-platform/opencti-front/src/private/components/analysis/malware_analyses/MalwareAnalysisDetails.tsx]: Jean-Philippe asked to change the shown name to Report name. Same for  Malware result -> Maliciousness

File: opencti-platform/opencti-front/src/private/components/observations/stix_cyber_observables/StixCyberObservableDetails.jsx [hunk 9]
```diff
@@ -146,16 +148,20 @@ const StixCyberObservableDetailsComponent = ({ stixCyberObservable }) => {
                   {t(observableAttribute.key.replace('attribute_', ''))}
                 </Typography>
                 <pre>
-                  <ItemCopy content={finalValue || '-'} />
+                  <ItemCopy content={finalValue || '-'}/>
```
Comment (SarahBocognano) [inline opencti-platform/opencti-front/src/private/components/observations/stix_cyber_observables/StixCyberObservableDetails.jsx]: Maybe you can use the FieldOrEmpty component ?

**Review by SarahBocognano — COMMENTED** [review]
Also noticed : 
- Point an observable in nested object of a malware analysis, click on that observable. The observable overview says that "No malware analysis on this observable" (I'm not sure it's the expected behavior)
- Export in txt empty
- Missing a lot of translation in Japanese and Chinese 

Capture d’écran (45)
Capture d’écran (46)
Capture d’écran (47)

File: opencti-platform/opencti-front/src/private/components/observations/stix_cyber_observables/StixCyberObservableDetails.jsx (hunk 9 above)
Comment (yassine-ouaamou) [reply opencti-platform/opencti-front/src/private/components/observations/stix_cyber_observables/StixCyberObservableDetails.jsx]: Out of the scope of this PR but you are definitely right

**Comment by yassine-ouaamou** [comment]
> Also noticed :
> 
> * Point an observable in nested object of a malware analysis, click on that observable. The observable overview says that "No malware analysis on this observable" (I'm not sure it's the expected behavior)
> * Export in txt empty
> * Missing a lot of translation in Japanese and Chinese
> 
> Capture d’écran (45) Capture d’écran (46) Capture d’écran (47)

@SarahBocognano 
1. Did you create a nested object with the sample reference and on a File/Url/Hostname/Domain-Name? -> @SarahBocognano My bad, I didn't put the sample reference... It's working perfectly fine with the proper test ^^
2. 🙃 
3. It was part of the first issue on Malware Analysis. I will fix that in a bug issue. Thanks!

**Comment by RomuDeuxfois** [comment]
Tests :
- Is not possible to have a relation sample between a malware analysis and a URL/domain name/shostname

**Comment by yassine-ouaamou** [comment]
> Tests :
> 
> * Is not possible to have a relation sample between a malware analysis and a URL/domain name/shostname

Indeed. It is not possible because we followed the Stix spec.
We agreed with @Jipegien and @Kedae to allow sample_ref on any kind of SCOs, which is not complient with Stix

**Comment by yassine-ouaamou** [comment]
All the feedbacks have been fixed. The export issue as txt is adressed in this PR: https://github.com/OpenCTI-Platform/connectors/pull/1214

**Comment by tomibennett** [comment]
> > Tests :
> > 
> > * Is not possible to have a relation sample between a malware analysis and a URL/domain name/shostname
> 
> Indeed. It is not possible because we followed the Stix spec. We agreed with @Jipegien and @Kedae to allow sample_ref on any kind of SCOs, which is not complient with Stix

erratum, only those observables can be referenced:
- file (stix supported)
- network traffic (stix supported)
- artifact (stix supported)
- domain name (not supported by stix)
- url (not supported by stix)
- hostname (not supported by stix)

As mentioned above, this cause inconsistency with the norm that are revealed upon export of a **malware analysis**. More info in issue #3538

**Pull request merged**